humminbird.cc

```diff
@@ -245,25 +247,22 @@ HumminbirdBase::humminbird_read_wpt(gbfile* fin)
 
   auto* wpt = new Waypoint;
 
-  // Could probably find a way to eliminate the alloc/copy.
-  char* s = xstrndup(w.name, sizeof(w.name));
-  wpt->shortname = s;
-  xfree(s);
+  QByteArray ba = QByteArray(w.name, sizeof(w.name));
+  wpt->shortname = QString(ba).remove(QChar::Null);
```

Comment:
Can this be
auto ba = QByteArray(w.name, qstrnlen(w.name, sizeof(w.name));
?
If so there are 2 more occurrences of this pattern below.

Reply:
or better
wpt->shortname = QByteArray(w.name, qstrnlen(w.name, sizeof(w.name));